pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -3029,6 +3076,7 @@ func TestEnsureExternalDNSDeployment(t *testing.T) {
 							BlockOwnerDeletion: &test.TrueVar,
 						},
 					},
+					Annotations: map[string]string{credentialsAnnotation: testSecretHash},
 				},
 				Spec: appsv1.DeploymentSpec{
 					Replicas: &replicas,
```

Comment:
Can you please add a test case when no secret existed and `ensureDeployment` failed due to this?